### PR TITLE
chore: Update DataFusion pin

### PR DESCRIPTION
**Pull request by alamb**
# Rationale
* Get upstream verification code https://github.com/apache/arrow-datafusion/pull/7333
* Keep up with dependencies

# Changes:
1. Update DataFusion pin to https://github.com/apache/arrow-datafusion/commit/f3722c0af8418bcb19cf9dc5f7e458a3aa5f0f32
2. Remove custom IOx verification added in https://github.com/influxdata/influxdb_iox/pull/8515 that I upstreamed

### CODE REVIEW

File: iox_query/src/exec/context.rs [hunk 1]
```diff
@@ -346,10 +342,13 @@ impl IOxSessionContext {
     pub async fn sql_to_logical_plan(&self, sql: &str) -> Result<LogicalPlan> {
         let ctx = self.child_ctx("sql_to_logical_plan");
         debug!(text=%sql, "planning SQL query");
-        // NOTE can not use ctx.inner.sql() here as it also interprets DDL
         let plan = ctx.inner.state().create_logical_plan(sql).await?;
-        // TODO use better API: https://github.com/apache/arrow-datafusion/issues/7328
-        verify_plan(&plan)?;
+        // ensure the plan does not contain unwanted statements
```
Comment (alamb) [inline iox_query/src/exec/context.rs:346]: this code was upstreamed -- the checks are covered by the tests in  https://github.com/influxdata/influxdb_iox/blob/53a2ece7b3f413b846089e3917ca392ad04de292/influxdb_iox/tests/query_tests/sql_errors.rs#L107-L158

File: iox_query/src/exec/context.rs [hunk 2]
```diff
@@ -346,10 +342,13 @@ impl IOxSessionContext {
...
+        let verifier = SQLOptions::new()
+            .with_allow_ddl(false) // no CREATE ...
+            .with_allow_dml(false) // no INSERT or COPY
+            .with_allow_statements(false); // no SET VARIABLE, etc
+        verifier.verify_plan(&plan)?;
```
Comment (NGA-TRAN) [inline iox_query/src/exec/context.rs:351]: 👍

**Review by NGA-TRAN — APPROVED** [review]
Nice and simpler verification

**Comment by alamb** [comment]
Thanks for the review @NGA-TRAN

**Pull request merged**